include/nuttx/fs/fs.h

```diff
@@ -224,6 +224,20 @@ struct geometry
   blksize_t geo_sectorsize;   /* Size of one sector */
 };
 
+struct partition_info_s
+{
+  uint32_t        magic;        /* File system magic, 0 for RAW
+                                 * (see <sys/statfs.h>) */
+  size_t          length;       /* Partition size in bytes */
+  size_t          numsectors;   /* Number of sectors in the partition */
+  size_t          sectorsize;   /* Size in bytes of a single sector */
+  off_t           startsector;  /* Offset to the first section/block of the
+                                 * managed sub-region */
+  off_t           endsector;    /* Offset to the last section/block of the
+                                 * managed sub-region */
+  FAR const char *parent;       /* Name of the parent node of the partition */
```

Comment:
change to array, otherwise userspace can't access the kernel pointer

Reply:
Good catch. Done.